shell/platform/darwin/ios/framework/Source/FlutterSpellCheckPlugin.mm

```diff
@@ -38,7 +38,13 @@ - (void)handleMethodCall:(FlutterMethodCall*)call result:(FlutterResult)result {
   NSString* method = call.method;
   id args = call.arguments;
   if ([method isEqualToString:kInitiateSpellCheck]) {
-    result([self initiateSpellCheck:args]);
+    FlutterError* error;
+    NSDictionary* spellCheckResult = [self initiateSpellCheck:args withError:&error];
+    if (error) {
```

Comment:
It's more idiomatic in Objective-C to check if the return value is nil, and then process the error.

Reply:
A returned nil value is valid without an error in our case here: that is when spell-check didn't find a misspelled word.
So if we check nil first, it will be:
```objc
if (! spellCheckResult && error) {
...
} else {
...
}
```

That's essentially the same with an extra unnecessary check.

Reply:
Or maybe `initiateSpellCheck:` should return something else to indicate a spell check didn't find a misspelled word? 

Since `NSNotFound` is not defined by us, it would be error prone to return `NSNotFound` to dart and rely on that. Maybe we can use -1 as the startingIndex when no misspelled word is found? So:
```
{
  startIndex: -1
  endIndex: -1
  suggestions: []
}
```

This way, `initiateSpellCheck:` only returns nil when there's an error, which better matches Objective-C idiomatic. WDYT

Reply:
> A returned nil value is valid without an error in our case here: that is when spell-check didn't find a misspelled word.

Can we make that case return an empty dictionary instead?